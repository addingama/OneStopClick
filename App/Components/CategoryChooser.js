import React, { Component, PropTypes } from 'react'
import { List, ListItem } from 'react-native-elements'

export default class CategoryChooser extends Component {
  generateItems (items) {
    const { selectedValue, onValueChange } = this.props
    return items.map(function (item) {
      return (
        <ListItem
          hideChevron={selectedValue !== item.name}
          key={item.id}
          title={item.name}
          onPress={() => onValueChange(item.name)}
          rightIcon={{name: 'check-circle', color: 'green'}}
        />
        // <Picker.Item label={item.name} value={item.name} key={item.id} />
      )
    })
  }
  render () {
    const { items } = this.props
    return (
      <List>
        { this.generateItems(items) }
      </List>
    )
  }
}

CategoryChooser.propTypes = {
  selectedValue: PropTypes.string.isRequired,
  items: PropTypes.array.isRequired,
  onValueChange: PropTypes.func.isRequired
}
