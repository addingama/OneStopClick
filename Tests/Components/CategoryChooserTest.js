import 'react-native'
import React from 'react'
import CategoryChooser from '../../App/Components/CategoryChooser'
import renderer from 'react-test-renderer'
const items = [
  {
    id: 1,
    name: 'All Category'
  },
  {
    id: 2,
    name: 'Movie'
  }
]
test('CategoryChooser component renders correctly', () => {
  const tree = renderer.create(<CategoryChooser selectedValue='Movie' items={items} onValueChange={() => alert('changed')} />).toJSON()
  expect(tree).toMatchSnapshot()
})
