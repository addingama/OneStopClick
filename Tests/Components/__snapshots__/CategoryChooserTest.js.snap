// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryChooser component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#bbb",
        "borderTopWidth": 1,
        "marginTop": 20,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderBottomColor": "#bbb",
            "borderBottomWidth": 1,
            "paddingBottom": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          },
          undefined,
        ],
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "marginLeft": 10,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          style={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "#43484d",
                    "fontSize": 17.5,
                  },
                  Object {
                    "marginLeft": 10,
                  },
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            All Category
          </Text>
        </View>
        <View
          style={undefined}
        >
          <View />
        </View>
      </View>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderBottomColor": "#bbb",
            "borderBottomWidth": 1,
            "paddingBottom": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          },
          undefined,
        ],
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "marginLeft": 10,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          style={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "#43484d",
                    "fontSize": 17.5,
                  },
                  Object {
                    "marginLeft": 10,
                  },
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            Movie
          </Text>
        </View>
        <View
          style={undefined}
        >
          <View />
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "flex-end",
            "flex": 0.15,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          onPress={undefined}
          style={
            Array [
              false,
              false,
              false,
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
          underlayColor="white"
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "green",
                  "fontSize": 28,
                },
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  undefined,
                ],
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            î¡¬
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
